# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/client/TextureRecorded.cpp
# Commit: 2b56c2b2837a
# Full Hash: 2b56c2b2837adfd7c49b599771cfea2d9e3d687b
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2023-11-22 21:50:37
# Regressor Bug: 1863914
# File Overlap Count: 2
# Description:
#   Bug 1863914: Use multiple shmem buffers for remote canvas recording. r=aosmond
#   
#   This replaces the use of a single large ring buffer.
#   The buffers are still processed in parallel and are recycled to reduce
#   allocation. Events that do not fit in the default sized buffer have a separate
# ==============================================================================

diff -r ed311995cd70 -r 2b56c2b2837a gfx/layers/client/TextureRecorded.cpp
--- a/gfx/layers/client/TextureRecorded.cpp	Wed Nov 22 14:06:43 2023 +0200
+++ b/gfx/layers/client/TextureRecorded.cpp	Wed Nov 22 11:52:08 2023 +0000
@@ -23,7 +23,7 @@
     already_AddRefed<CanvasChild> aCanvasChild, gfx::IntSize aSize,
     gfx::SurfaceFormat aFormat, TextureType aTextureType)
     : mCanvasChild(aCanvasChild), mSize(aSize), mFormat(aFormat) {
-  mCanvasChild->EnsureRecorder(aTextureType);
+  mCanvasChild->EnsureRecorder(aSize, aFormat, aTextureType);
 }
 
 RecordedTextureData::~RecordedTextureData() {